### ISSUE TITLE
[Recommendation] Suggestions for handling the fluid animation problem

### ISSUE THREAD

**Issue by yxliang01** [BuildCraft/BuildCraft#4113]
I feel like buildcraft should either disable fluid animation in the new builds until the animation problem is confirmed fixed, or in the versions which have the animation problem, show a message to state that "if it's lagging, please considering downloading the static fluid resource pack". This message potentially prevents issues like #3961 #4064 which are just stating the same known issues and makes buildcraft more usable. It's really difficult to know that the lagging problem is from the buildcraft without this kind of message and you have lots of mods.

Thanks

**Comment by AEnterprise** [comment]
not sure that would help much, people will generally jump straight to using it forever then and whenever something does lag blame it all on BC, even after the fix

**Comment by yxliang01** [comment]
Hmm.. How about disable the animation by default until it's confirmed fixed?

**Comment by AEnterprise** [comment]
not sure if that's worth it considering from the few reports we've seen this only seems to impact a minority of the users and this would be taking away the feature from everyone

**Comment by AlexIIL** [comment]
Having an option to disable the behaviour and fall-back on something other than a missing texture is basically the original fix...

On second thought it shouldn't take too long to add a config option to disable animations for out textures.

**Comment by yxliang01** [comment]
Yes. It would be great to have the option. Meanwhile, I think adding the hint to players who are seemed to be lagging would also be helpful. Thanks!

**Comment by AlexIIL** [comment]
Ok, so that's the config option added under a new "performance" category. I'm not quite so sure about the hint.

**Comment by AlexIIL** [comment]
(This partially implements a fix for #3616)

**Comment by AlexIIL** [comment]
Ideally we could auto-disable it if it's taking a lot of tick time... But I'm not sure how to accurately measure all of them.

**Comment by yxliang01** [comment]
OMG Thanks a lot @AlexIIL ! That was quick. The reason why I made this issue is because I wasn't sure why it was so laggy for several months already. If the hint was there, I could have known that I'm supposed to try disabling the animation back to that time.

**Comment by AlexIIL** [comment]
Can you quickly test this with [7.99.16-pre4](https://www.mod-buildcraft.com/releases/testing/BuildCraft/7.99.16-pre4/buildcraft-7.99.16-pre4.jar)? Just to ensure this actually fixes the issue (/gives a good enough hint).

WARNING: this is 7.99.16 which *might* have breaking changes in. I'm not sure atm :/

**Comment by yxliang01** [comment]
@AlexIIL Thanks for adding the hint. Just tested it, both the hint and the config worked! I think this issue can be closed now.

Thanks!

**Comment by AlexIIL** [comment]
Ok. I'll close the issue when 7.99.16 is actually released though :)
